### PR TITLE
Various Android fixes

### PR DESCRIPTION
**Pull request by dead10ck**
# Description

#10013 got Android compiling successfully, but there are a few remaining issues that this PR fixes.

* Trash
  * Fix unused variable warnings on Android
  * Add a convenience configuration option called `trash` to signal when the `trash-support` feature is enabled and the build machine is one of the platforms which has trash
* Some unit tests fail
  * The path to the binaries for tests is slightly incorrect. It is missing the build target when it is set with the `CARGO_BUILD_TARGET` environment variable. It is common on Termux to set `CARGO_BUILD_TARGET` explicitly since the default target that rustc detects can cause problems on some projects, such as python's `cryptography` package.
  * Additionally, the existing variable named `NUSHELL_CARGO_TARGET` is in fact the profile, not the build target, so this was corrected.
  * `std path add` tests were missing `android` test
* upgrade nix to 0.27
  * This fixes a segfault on Android when fetching the user group. See: https://github.com/nix-rust/nix/issues/2084

# User-Facing Changes

None

# Tests + Formatting

All tests now pass on Android on Termux

### CODE REVIEW

File: build.rs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+fn main() {
```
Comment (jntrnr) [inline build.rs:1]: Can we do this without adding a build.rs file just for this purpose?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -7,7 +7,7 @@ on:
 name: continuous-integration
 
 env:
-  NUSHELL_CARGO_TARGET: ci
+  NUSHELL_CARGO_PROFILE: ci
```
Comment (jntrnr) [inline .github/workflows/ci.yml:10]: Can we make this a separate PR as it's unrelated to your change?

File: crates/nu-test-support/src/fs.rs [hunk 3]
```diff
@@ -233,18 +233,21 @@ pub fn root() -> PathBuf {
...
+    };
 
     std::env::var("CARGO_TARGET_DIR")
-        .ok()
-        .map(|target_dir| PathBuf::from(target_dir).join(&build_type))
-        .unwrap_or_else(|| root().join(format!("target/{}", &build_type)))
+        .map(PathBuf::from)
+        .unwrap_or_else(|_| root().join("target"))
+        .join(build_target)
+        .join(profile)
```
Comment (jntrnr) [inline crates/nu-test-support/src/fs.rs:250]: I'm a bit unclear on how this fix is related to the platform support.

For unrelated changes, we generally do separate, targeted fixes (just to make things a bit easier to review)

**Review by jntrnr — CHANGES_REQUESTED** [review]

**Comment by jntrnr** [comment]
Thanks for the fixes for Android. I use termux on my phone, so I'd definitely like to improve the support in Nushell itself.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (dead10ck) [reply .github/workflows/ci.yml:10]: I'm not sure that it's unrelated; there is a need to use a variable for the build target, and a variable for the build profile. Without this change, the build system would be using `CARGO_BUILD_TARGET` to specify the build target and `NUSHELL_CARGO_TARGET` to specify the build profile, which is pretty confusing. I'll change it back if you really want though.

File: build.rs (hunk 1 above)
Comment (dead10ck) [reply build.rs:1]: I'm open to suggestions. The reason I added this was because @kubouch gave feedback in [the last PR](https://github.com/nushell/nushell/pull/10013#pullrequestreview-1596828128) that it was unwieldy to repeat

```rust
    #[cfg(all(
        feature = "trash-support",
        not(target_os = "android"),
        not(target_os = "ios")
    ))]
```

in half a dozen places, which I agree with. Another option would be [`cfg_aliases`](https://docs.rs/cfg_aliases/latest/cfg_aliases/), but I opted not to go this route because of other feedback about not adding more external dependencies.

I'm open to anything though, this was really just added in response to other review feedback.

File: crates/nu-test-support/src/fs.rs (hunk 3 above)
Comment (dead10ck) [reply crates/nu-test-support/src/fs.rs:250]: I explained this in the PR description—when the `CARGO_BUILD_TARGET` is set explicitly (which is common on Termux due to the increasing use of Rust in Python packages), it is part of the build output path, so e.g. on my device, the path to the `nu` binary is

`target/aarch64-linux-android/debug/nu`

rather than just

`target/debug/nu`

It's true that this is more general of an issue than just Android support, but I think more likely on Android due to happenstance.

I can split this off into another PR if you'd like though.

**Comment by dead10ck** [comment]
In general I'm happy to split this into smaller PRs if that's desired—I totally understand it's more difficult to review these "kitchen sink" style PRs.

In this case, each of the changes was very small and relatively self contained, so I opted to just put them all together. If it helps, I intentionally made a self contained commit for each of the changes, so reviewing each commit might make it easier. Just let me know which parts you'd like me to split off, considering the other feedback I gave.

**Comment by jntrnr** [comment]
@dead10ck - yeah if you don't mind splitting up the PR into related changes, that'll be easier. that way other folks can jump in and review the other PRs.

**Comment by dead10ck** [comment]
Closing in favor of:

#10223
#10224
#10225
